test/integration/policy_selector_test.go

```diff
@@ -0,0 +1,230 @@
+			},
+		}
+
+		testEntries := []TableEntry{}
+		// Populate the test entries
+		for name, test := range tests {
+			testEntries = append(testEntries, Entry("With objectSelector with "+name, test))
+		}
+
+		// Base test definition
+		baseSelectorTest := func(test selectorTest) {
+			It(policyName+" should be created on the Hub", func(ctx SpecContext) {
+				common.DoCreatePolicyTest(policyYAML, common.GvrConfigurationPolicy)
+
+				By("Patching the objectSelector")
+				selector, err := json.Marshal(test.selector)
+				Expect(err).NotTo(HaveOccurred())
+				jsonPath := "/spec/policy-templates/0/objectDefinition/" +
+					"spec/object-templates/0/objectSelector/matchExpressions"
+				_, err = common.OcHub(
+					"patch", "policies.policy.open-cluster-management.io", policyName,
+					"-n", userNamespace, "--type=json", "-p", `[{
+					"op":"replace", 
+					"path":"`+jsonPath+`",
+					"value": `+string(selector)+`}]`)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It(policyName+" should have the expected status", func(ctx SpecContext) {
+				Eventually(common.GetLatestStatusMessage(policyName, 0),
+					defaultTimeoutSeconds, 1).Should(Equal(test.status))
+			})
+		}
+
+		///////////////////
+		// Run the tests //
+		///////////////////
+
+		BeforeAll(func(ctx SpecContext) {
+			By("The ConfigMaps should be created on the Managed cluster")
+			By(fmt.Sprintf("Create namespace %s on the Managed cluster", configNamespace))
+			namespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{Name: configNamespace},
+			}
+
+			_, err := clientManaged.CoreV1().Namespaces().Create(ctx, namespace, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			for i, name := range configMapNames {
+				By(fmt.Sprintf("Creating ConfigMap %s/%s on the Managed cluster", configNamespace, name))
+				configMap := &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   name,
+						Labels: map[string]string{selectorKey: name},
+					},
+					Data: map[string]string{"name": fmt.Sprintf("%s%d", "testvalue", i)},
+				}
+
+				_, err := clientManaged.CoreV1().ConfigMaps(configNamespace).Create(
+					ctx, configMap, metav1.CreateOptions{})
+				if !k8serrors.IsAlreadyExists(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+
+				By(fmt.Sprintf("Creating ConfigMap %s/%s on the Managed cluster", configNamespace, unrelatedConfigMap))
+				_, err = clientManaged.CoreV1().ConfigMaps(configNamespace).Create(ctx, &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: unrelatedConfigMap,
+					},
+				}, metav1.CreateOptions{})
+				if !k8serrors.IsAlreadyExists(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+		})
+
+		DescribeTableSubtree("Reporting the correct status", baseSelectorTest, testEntries)
+
+		Describe("Reporting correct status with object updates", func() {
+			// Run a predefined base set of checks, including applying the policy
+			baseSelectorTest(defaultTest)
```

Comment:
Does this mean that default test is run twice? I think that's ok, I guess the side-effects are useful here?

Reply:
Yes--it includes applying the policy. Basically kind of a reset before moving on to other checks.

Reply:
Though that said, I could do a refactor to have a `BeforeEach` to apply the policy instead...